dbadapter/db_adapter.go

```diff
@@ -56,6 +56,33 @@ type MongoDBClient struct {
 	mongoapi.MongoClient
 }
 
+func (db *MongoDBClient) Client() *mongo.Client {
+	return db.MongoClient.Client
+}
+
+type SessionRunner func(ctx context.Context, fn func(sc mongo.SessionContext) error) error
+
+func RealSessionRunner(client *mongo.Client) SessionRunner {
```

Comment:
```suggestion
func RealSessionRunner(client DBInterface) SessionRunner {
```
I think this could work because the `DBInterface` already implements the `StartSession`, like this we do not have to retrieve the client and pass it to the method , we could just do
 ```
 sessionRunner := dbadapter.RealSessionRunner(dbadapter.CommonDBClient)
 ```